### ISSUE TITLE
Resetting of the riverstates in _cluster

### ISSUE THREAD

**Issue by marcelalburg** [jprante/elasticsearch-river-jdbc#491]
Hello,

how can i reset the riverstates in _cluster/state?pretty ?

In my opinion there should be an param. if i create (or override) a river the param can set to "true" and resets the riverstates for the river.

Maybe like this:

`
curl -XPUT '127.0.0.1:9200/_river/testriver/_meta' -d '{
  "type": "jdbc",
  "reset_ riverstates": true
 .
 .
 .
}
`
Thanks
Marcel

**Comment by jprante** [comment]
The state is immutable, you have to recreate a river if you want to reset it.

**Comment by marcelalburg** [comment]
Yeah, but i recognize if i delete and (re)create the river again (with the same name). the riverstates are not reseted.

**Comment by jprante** [comment]
That is a bug, I will have a look at this. Thanks for reporting.

**Comment by marcelalburg** [comment]
Hello,

any news about this ?

Thanks

**Comment by jprante** [comment]
It is fixed in this commit https://github.com/jprante/elasticsearch-river-jdbc/commit/b3a913cf06dba6fe91f43a16796bdfdf93bc82bc

**Comment by mickremy** [comment]
Is it possible to not delete the river state when elasticsearch sutdown ?

**Comment by marcelalburg** [comment]
Ohh ok, but i use currently ES 1.4.x. it is possible to merge to 1.4.x branch ?

Thanks
Marcel

**Comment by jprante** [comment]
Sorry, the 1.4 branch commit is https://github.com/jprante/elasticsearch-river-jdbc/commit/9252a5e8f465c636560d5be53ec13f03e93d3456

It is available in JDBC plugin 1.4.4.0

**Comment by marcelalburg** [comment]
Ohh ok. Perfect. I'll test this and close this ticket, if it works.

Thanks for your Work
Marcel

**Comment by jprante** [comment]
@mickremy this was considered a bug. I will overhaul the state management in a new version.

**Comment by mickremy** [comment]
Ok thanks @jprante.
I would like too help you with that however I don't know where to start, any idea ?
